### PR TITLE
Subscription management: Refactor SiteSubscriptionsList to allow for custom empty state, and not found, components

### LINKED ISSUE #78631: Subscription Management: Add ability to search to discover new sites and ability to add subscription by URL.
When a person enters a keyword in the search box, it will do both:
 - filter the existing subscription list
 - show a list of new sites to discover

The search box should also support pasting of URLs for:
- Wordpress Site URL.
- RSS URL

where the entry of the pasted URL will be shown along with the list of new sites to discover.

Prioritize making this work on Reader portal first.

The design will be similar to the recommended sites' design, actual design will be clarified soon.

Bear in mind the design will also include how it will work with paid sites. For now, let's get the backbone built first.

image

### PR DESCRIPTION
**Pull request by mashikag**
Fixes #78631

## Proposed Changes

* There is no visual changes (kinda)
   * I took the opportunity to fix the layout of the sorts component on the Comments subscriptions page  

| Before | After |
|:---:|:---:|
|  |  |

* Refactor the `SiteSubscriptionsList` component to allow for empty state (`emptyComponent` prop) and not found state (`notFoundComponent` prop)
   * **With the new design for the subscriptions list in the reader portal, the two lists will need to display empty and not found states differently**
* Now also, that the `SiteSubscriptionsQueryContext` was separated out from `useSiteSubscriptionsQuery` in [the previous PR](https://github.com/Automattic/wp-calypso/pull/78896), it makes sense to get rid of the `SiteSubscriptionManager` component, and instead export `SiteSubscriptionsList` and `SiteSubscriptionsListActionsBar` on their own - without the need for the `SiteSubscriptionManager` compound component

## Testing Instructions

* Ensure there are no regressions to site subscriptions fetching in both:
   * the reader portal - http://calypso.localhost:3000/read/subscriptions
   * the external subscriptions portal - http://calypso.localhost:3000/subscriptions/sites

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/landing/subscriptions/components/tab-views/comments/styles.scss [hunk 1]
```diff
@@ -1,5 +1,5 @@
 @import "client/landing/subscriptions/styles/list-actions-bar";
 
-.tab-view {
```
Comment (roo2) [inline client/landing/subscriptions/components/tab-views/comments/styles.scss:3]: I can see that in `client/landing/subscriptions/components/tab-views/tab-view.tsx` we are still using the `.tab-view` class. I think this needs to be updated as well.

File: client/landing/subscriptions/styles/list-actions-bar.scss [hunk 2]
```diff
@@ -1,52 +1,51 @@
 @import "@wordpress/base-styles/breakpoints";
 
 %list-actions-bar {
-	.list-actions-bar {
```
Comment (roo2) [inline client/landing/subscriptions/styles/list-actions-bar.scss:4]: I suspect that taking this class out might affect `client/my-sites/subscribers/components/subscriber-list-actions-bar/subscriber-list-actions-bar.tsx` which still uses the `.list-actions-bar` class.

**Review by phcp — APPROVED** [review]
Great job @mashikag! No regressions noticed on my tests.

File: client/landing/subscriptions/components/tab-views/comments/styles.scss (hunk 1 above)
Comment (mashikag) [reply client/landing/subscriptions/components/tab-views/comments/styles.scss:3]: Yes, we are still using the `.tab-view` class in `client/landing/subscriptions/components/tab-views/tab-view.tsx`. The styles defined in this `styles.scss` were meant to be applied to the `.tab-view` class in the `comments` tab-view component. To be fair, the whole `TabView` component seems like an unnecessary wrapper component that imposes restrictions on its children. I can see the `TabView` component being removed, once we get to re-visit the `Comments` view.

File: client/landing/subscriptions/styles/list-actions-bar.scss (hunk 2 above)
Comment (mashikag) [reply client/landing/subscriptions/styles/list-actions-bar.scss:4]: Actually, the `.list-actions-bar` has its own style definition in `client/my-sites/subscribers/components/subscriber-list-actions-bar/style.scss` (see [here](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/subscribers/components/subscriber-list-actions-bar/style.scss)). 🙈 I tested it out quickly, and my PR branch does not regress the behavior of the subscribers page from staging. 

However, thanks for flagging this @roo2 ! @JuanLucha, this is probably something you should be aware of. Using the same class names for different components may run us into trouble. Similar to what I faced recently with the site-icon: p1687769977039419-slack-C9EJ7KSGH. 😅

**Pull request merged**